src/pages/home/sidebar/ChatSwitcherView.js

```diff
@@ -160,7 +160,14 @@ class ChatSwitcherView extends React.Component {
             .map((report) => {
                 const participants = lodashGet(report, 'participants', []);
                 const isSingleUserDM = participants.length === 1;
+                const isGroupDM = participants.length > 1;
+                const personalDetails = this.props.personalDetails || {};
+                const participant1Avatar = personalDetails[participants[0]]
+                    ? personalDetails[participants[0]].avatarURL : '';
+                const participant2Avatar = personalDetails[participants[1]]
+                    ? personalDetails[participants[1]].avatarURL : '';
```

Comment:
We could build the avatar URLs this way. But we already have some logic which does this elsewhere.

https://github.com/Expensify/ReactNativeChat/blob/bf4926b93d884aa41d942a3e351485931cc5ebbc/src/libs/actions/PersonalDetails.js#L164-L173

This logic runs each time a report is loaded. Perhaps if we simply changed `icon` to `icons` we would not have to leave this logic in two separate places.

Reply:
Thanks for this. I guess this is what @sketchydroide  was referring to in his comment [here](https://github.com/Expensify/Expensify.cash/pull/1105#discussion_r549751554).
I've fixed it. Please confirm